s3/pkg/datastore/aws/aws.go

```diff
@@ -166,13 +218,56 @@ func (ad *AwsAdapter) Get(ctx context.Context, object *pb.Object, start int64, e
 	return result.Body, nil
 }
 
+
+func (ad *AwsAdapter) BucketDelete(ctx context.Context, in *pb.Bucket) error {
+	log.Info("Bucket delete is called in aws s3 service")
+	Credentials := credentials.NewStaticCredentials(ad.backend.Access, ad.backend.Security, "")
+	configuration := &aws.Config{
+		Region:      aws.String(DefaultRegion),
+		Endpoint:    aws.String(ad.backend.Endpoint),
+		Credentials: Credentials,
+	}
+
+	svc := awss3.New(session.New(configuration))
+	input := &awss3.DeleteBucketInput{
+		Bucket: aws.String(in.Name),
+	}
+
+	result, err := svc.DeleteBucketWithContext(ctx, input)
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			default:
+				log.Error(aerr.Error())
+			}
+		} else {
+			// Print the error, cast err to awserr.Error to get the Code and
+			// Message from an error.
+			log.Error(err.Error())
+		}
+		return err
+	}
+	log.Info(result)
+
+	return nil
+}
+
 func (ad *AwsAdapter) Delete(ctx context.Context, input *pb.DeleteObjectInput) error {
-	bucket := ad.backend.BucketName
-	objectId := input.Bucket + "/" + input.Key
+	log.Info("Bucket object delete is called in aws s3 service")
+	bucket := input.Bucket
+	objectId := input.Key
+	log.Info("Deleting the object of bucket:%s and object:%s", bucket, objectId)
 	deleteInput := awss3.DeleteObjectInput{Bucket: &bucket, Key: &objectId}
 
 	log.Infof("delete object[AWS S3], objectId:%s.\n", objectId)
-	svc := awss3.New(ad.session)
+
+	Credentials := credentials.NewStaticCredentials(ad.backend.Access, ad.backend.Security, "")
```

Comment:
The driver (AwsAdapter) has already created the session. Why to create it here again?

Reply:
But we need to create a new session everytime.
Ex: svc := awss3.**New**(ad.session)